BHoM_Engine/Compute/RecordEvent.cs

```diff
@@ -46,6 +46,31 @@ public static bool RecordEvent(string message, EventType type = EventType.Unknow
 
         /***************************************************/
 
+        [Description("Record an event with details of a C# exception within the BHoM logging system.")]
+        [Input("exception", "The C# exception being caught to provide the event and stack information for.")]
+        [Input("message", "An optional additional message which will be displayed first in the event log.")]
+        [Output("success", "True if the event has been successfully recorded as a BHoM Event.")]
+        public static bool RecordEvent(Exception exception, string message = "", EventType type = EventType.Unknown)
+        {
+            string exceptionMessage = "";
+
+            Exception e = exception;
+            while(e != null)
+            {
+                exceptionMessage += $"{e.Message}{Environment.NewLine}{Environment.NewLine}";
+                e = e.InnerException;
+            }
+
+            if (!string.IsNullOrEmpty(message))
+                message = $"{message}\n\n{exceptionMessage}";
+            else
+                message = exceptionMessage;
+
+            return RecordEvent(new Event { Message = message, StackTrace = exception.StackTrace, Type = EventType.Unknown });
```

Comment:
Just leaving here my personal opinion, nothing wrong with the PR. The mechanism that we currently have and that this PR is pushing is well working and well suited to UIs.
My personal preference would be to store the exception in the Event, and have the `RecordEvent` mechanism simply recording the exception itself, possibly with a message. When data extracted from the exception is needed (e.g. the StackTrace, or the concatenated InnerExceptions) we could have another method that does the necessary processing to the Event.Exception data as required by the target context (e.g. the UI). This would allow to deal with the recorded events in different ways in different contexts (i.e. console applications) and would also expose more exception data that may be useful to filter certain events if required by the target context.

Reply:
If we were to do this I would prefer it be a new issue as that would involve changing the event object itself and broadens the scope of this work considerably. I think that falls better into the logging discussions that I've had with @pawelbaran recently too and is probably one to discuss in more detail during our upcoming hack sessions, so if it's ok with you @alelom I'll mark this as resolved on the basis of being out of scope for this PR but one for us to discuss at the upcoming hack sessions (feel free to unresolve if you disagree!) 😄

Reply:
Absolutely! The comment was just to record my opinion and to have further discussions down the line. Thanks!